libraries/extensions/ros2-bridge/msg-gen/src/types/action.rs

```diff
@@ -132,6 +205,196 @@ impl Action {
+                            feedback_stream.for_each(|feedback| async {
+                                match feedback {
+                                    Ok(feedback) => println!("Received feedback: {:?}", feedback),
+                                    Err(e) => eprintln!("Error while receive feedback: {:?}", e),
+                                }
+                            }).await;
+                        }
+                    };
+
+                    self.executor.spawn(feedback_handle).context("failed to spawn feedback task")?;
+
+
+                    let status_handle = {
+                        let client_ref = Arc::clone(&client_arc);
+                        async move {
+                            let status_stream = client_ref.status_stream(goal_id);
+                            status_stream.for_each(|status| async {
+                                match status {
+                                    Ok(status) => println!("Status update: {:?}", status),
+                                    Err(e) => eprintln!("Error receiving status update: {:?}", e),
+                                }
+                            }).await;
+                        }
+                    };
+
+                    self.executor.spawn(status_handle).context("failed to spawn status task")?;
+
+                    let request_result_handle = {
+                        let client_ref = Arc::clone(&client_arc);
+                        let response_tx = self.response_tx.clone();
+                        async move {
+                            match client_ref.async_request_result(goal_id).await {
+                                Ok((_status, result)) => {
+                                    let response = Ok(result);
+                                    if response_tx.send_async(response).await.is_err() {
+                                        tracing::warn!("failed to send action result");
+                                    }
+                                },
+                                Err(e) => {
+                                    tracing::error!("Failed to receive response for request {goal_id:?}: {:?}", e);
+                                }
+                            }
+                        }
+                    };
+
+                    self.executor.spawn(request_result_handle).context("failed to spawn response task").map_err(|e| eyre::eyre!("{e:?}"))?;
+                    Ok(())
+                }
+
+                #[allow(non_snake_case)]
+                fn #matches(&self, event: &crate::ffi::CombinedEvent) -> bool {
+                    match &event.event.as_ref().0 {
+                        Some(crate::MergedEvent::External(event)) if event.id == self.stream_id => true,
+                        _ => false
+                    }
+                }
+
+                #[allow(non_snake_case)]
+                fn #downcast(&self, event: crate::ffi::CombinedEvent) -> eyre::Result<()> {
```

Comment:
this method should return some type,but if i add ffi:result_type_raw,it can't pass compile

Reply:
I opened https://github.com/starlitxiling/dora/pull/1 against your PR with a fix for this.